### PR TITLE
Spelling src etc

### PR DESCRIPTION
**Pull request by jsoref**
The various src/* items seem slightly disparate, so I'm doing src/* individually.

split from https://github.com/rust-lang/rust/pull/110392

### CODE REVIEW

File: src/etc/lldb_lookup.py [hunk 1]
```diff
@@ -115,4 +115,4 @@ def synthetic_lookup(valobj, dict):
     if rust_type == RustType.STD_REF_CELL:
         return StdRefSyntheticProvider(valobj, dict, is_cell=True)
 
-    return DefaultSynthteticProvider(valobj, dict)
+    return DefaultSyntheticProvider(valobj, dict)
```
Comment (jsoref) [inline src/etc/lldb_lookup.py:118]: This is the only interesting part of this change. I'm assuming that uppercase doesn't imply a frozen/public symbol.

**Review by jsoref — COMMENTED** [review]
I'll squash if the changes are ok

**Comment by Mark-Simulacrum** [comment]
Please squash commits, r=me after that.

**Pull request merged**